file/lib/src/sentry_io_overrides_integration.dart

```diff
@@ -0,0 +1,29 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:sentry/sentry.dart';
+import 'sentry_io_overrides.dart';
+
+/// When installed, every new file will be created as [SentryFile].
+/// When installed, operations will use [SentryFile] instead of dart:io's [File]
+/// implementation whenever [File] is used.
+class SentryIOOverridesIntegration extends Integration<SentryOptions> {
+  IOOverrides? _previousOverrides;
+  SentryOptions? _options;
+  @override
+  FutureOr<void> call(Hub hub, SentryOptions options) {
+    _options = options;
+    if (options.isTracingEnabled()) {
+      _previousOverrides = IOOverrides.current;
```

Comment:
@marandaneto I'm not sure if we can even store & restore the correct override. Since we can only read `current`, which might not be the `global` one, but we can only write to the `global` one directly. Any advice?

Reply:
@denrase the goal here is to restore the previous value, if it's already overridden by any other library, that's fine.

Reply:
I've additionally documented the current behaviour.